### PR TITLE
Use flux boundary condition for tke

### PR DESCRIPTION
**Pull request by costachris**
Replace Dirichlet boundary condition for tke with flux boundary condition.

### CODE REVIEW

File: src/EDMF_functions.jl [hunk 1]
```diff
@@ -962,6 +964,20 @@ function compute_en_tendencies!(
...
+    # compute bottom BC for TKE
+    if is_tke
+        ρ_f_surf = ρ_f[kf_surf]
+        a_e_surf = 1 - edmf.surface_area
+        u_surf = physical_grid_mean_u(state)[kc_surf]
+        v_surf = physical_grid_mean_v(state)[kc_surf]
+        U_surf_norm = sqrt(u_surf^2 + v_surf^2)
+        ustar = surf.ustar
+        surface_tke_turb_flux = get_surface_tke_turb_flux(mix_len_params, ustar, ρ_f_surf, a_e_surf, U_surf_norm)
+        ∇c_turb = CCO.DivergenceF2C(; bottom = CCO.SetValue(CCG.Covariant3Vector(surface_tke_turb_flux)))
```
Comment (szy21) [inline src/EDMF_functions.jl:976]: I think it should be a `CCG.WVector` here based on your function that calculate `surface_tke_turb_flux`, but then I am surprised it works...

File: src/turbulence_functions.jl [hunk 2]
```diff
@@ -58,6 +58,14 @@ function get_surface_tke(mixing_length_params, ustar::FT, zLL::FT, oblength::FT)
         return κ_star² * ustar * ustar
     end
 end
+
+function get_surface_tke_turb_flux(mixing_length_params, ustar::FT, ρ_f_surf::FT, a_e_surf::FT, U_surf_norm::FT) where {FT}
+    κ_star² = mixing_length_params.κ_star²
```
Comment (szy21) [inline src/turbulence_functions.jl:63]: Is κ_star² von karman constant squared?

File: src/EDMF_functions.jl (hunk 1 above)
Comment (costachris) [reply src/EDMF_functions.jl]: that was based on analogous operations in ClimaAtmos with DivergenceF2C, but I'll use WVector instead

File: src/turbulence_functions.jl (hunk 2 above)
Comment (costachris) [reply src/turbulence_functions.jl:63]: `κ_star²` is ratio of TKE to squared friction velocity. (`κ_star` is ratio of rms turbulent velocity to friction velocity)

Comment (szy21) [reply src/turbulence_functions.jl:63]: Ah, I might do it wrong in CilmaAtmos then. Is that parameter in https://github.com/CliMA/CLIMAParameters.jl/blob/main/src/parameters.toml?

**Review by szy21 — APPROVED** [review]

**Pull request merged**